src/hotspot/share/runtime/thread.cpp

```diff
@@ -482,16 +482,17 @@ void Thread::check_for_dangling_thread_pointer(Thread *thread) {
 }
 #endif
 
-// Is the target JavaThread protected by this Thread:
+// Is the target JavaThread protected by the calling Thread:
```

Comment:
s/calling/current/ ?

We could also expand this to clarify that even if a target JavaThread is not explicitly protected by the current thread it can still be safe to access if the two threads engage in a suitable protocol.